proposals/20190909-psp-rules-support.md

```diff
@@ -0,0 +1,50 @@
+# Support for K8s Pod Security Policies (PSPs) in Falco
+
+<!-- toc -->
+
+- [Summary](#summary)
+- [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  * [Use cases](#use-cases)
+  * [Diagrams](#diagrams)
+  * [Design Details](#design-details)
+
+<!-- tocstop -->
+
+## Summary
+
+We want to make it easier for K8s Cluster Operators to Author Pod Security Policies by providing a way to read a PSP, convert it to a set of falco rules, and then run falco with those rules.
+
+## Motivation
+
+PSPs provide a rich powerful framework to restrict the behavior of pods and apply consistent security policies across a cluster, but it’s difficult to know the gap between what you want your security policy to be and what your cluster is actually doing. Additionally, since PSPs enforce once applied, they might prevent pods from running, and the process of tuning a PSP live on a cluster can be disruptive and painful.
```

Comment:
How we guarantee that our mapping is always 1:1 between what kubernetes does and what falco does?

Reply:
It's a risk, but I tried really hard to be comprehensive and added a big set of tests to verify all the parts of the generated rules match the activity covered by the PSP. You can see the tests I wrote at https://github.com/falcosecurity/falco/blob/psp-rules-support/test/falco_tests_psp.yaml.

Reply:
OK thanks for clarifying